### ISSUE TITLE
Unable to get credentials, even with non microsoft account

### ISSUE THREAD

**Issue by spuder** [pendrica/azure-credentials#4]
This gem only works with non microsoft accounts. As a workaround suggested in #1  I've made a new user in the active directory account with global admin rights. 

![screenshot 2016-10-06 10 33 49](https://cloud.githubusercontent.com/assets/242382/19161582/93a2da10-8bb1-11e6-9879-fbdd71411a62.png)

When I run azure-credentials with this new user, I no longer get the error `Unable to acquire token from Azure AD provider`, however the gem doesn't return the credentials I would expect. 
Logging into the new portal doesn't show any new role assignments

```
[sowen:/tmp] $ azure-credentials -u xxxxxxx -p 'xxxxxxx' -o /tmp/tf.creds -t terraform -v
INFO [2016-10-06 10:36:03] Authenticating to Azure Active Directory
INFO [2016-10-06 10:36:03] Retrieving tenant info
INFO [2016-10-06 10:36:04] Retrieving subscriptions info
INFO [2016-10-06 10:36:04] Creating credentials file at /tmp/tf.creds

INFO [2016-10-06 10:36:04] Done!
[sowen:/tmp] $ cat /tmp/tf.creds
[sowen:/tmp] $
```

I'm really unfamiliar with azure, so any guidance of where to go from here would be appreciated.

**Comment by stuartpreston** [comment]
Hi @spuder - whilst your domain user is a global admin, you still need to add the user as a "co-administrator for the subscription" in order for the correct tenant to be determined by the tool.  In the "old portal" that you are in, navigate to Settings on the left side then Administrators

![image](https://cloud.githubusercontent.com/assets/6384654/19161896/84730e5c-8bed-11e6-92f3-77b5cb6dfd12.png)

**Comment by spuder** [comment]
Thanks for the tips. Microsoft sure doesn't make this easy. It warns that I'm not meeting 2 criteria. 

As far as I can tell, both these conditions are true. 

The co-administrator is a user in the NXXXX Directory
I am also a user in that directory

Perhaps I need to talk with microsoft support. Any other suggestions before I go bug them? 

![screenshot 2016-10-06 10 57 52](https://cloud.githubusercontent.com/assets/242382/19162172/11969cd4-8bb4-11e6-9d90-8d3d140ad1f4.png)

**Comment by stuartpreston** [comment]
I can't think of anything specific, have you attempted to login once with this user already (to change the password) - in a private window at http://myapps.microsoft.com 

Otherwise I think I'd recommend following the azure cli steps at: https://azure.microsoft.com/en-gb/documentation/articles/resource-group-authenticate-service-principal-cli/ to get your service principal set up...

**Comment by stuartpreston** [comment]
...and don't forget there's a docker image for the CLI: ```microsoft/azure-cli```

**Comment by spuder** [comment]
Still trying to shave this yak. 

You state in the other issue: 

> I just tested using an account that was setup with a Microsoft Account as the Global Administrator, and added a new AAD account as a co-administrator:

Is an AAD account is different than an active directory account?

**Comment by stuartpreston** [comment]
No it's the same thing. Also remember you can add -l debug into the azure-credentials command to get some more info about authentication (potentially).

**Comment by spuder** [comment]
I'm stumped. Its almost like microsoft doesn't want you to use this feature. Debug isn't showing me much. 

```bash
azure-credentials -u testuser@xxxxxx.onmicrosoft.com -o /tmp/tf.creds -t terraform -v -l debug

DEBUG [2016-10-07 20:59:33] Command line options: {:subscription_id=>nil, :role=>"Contributor", :type=>"terraform", :log_level=>:debug, :output_file=>"/tmp/tf.creds", :out_to_screen=>true, :username=>"testuser@xxxxxxxx.onmicrosoft.com"}
Enter your password: 
INFO [2016-10-07 20:59:45] Authenticating to Azure Active Directory
DEBUG [2016-10-07 20:59:46] Request: https://login.windows.net/Common/oauth2/token (POST) resource=https%3A%2F%2Fmanagement.core.windows.net%2F&client_id=xxxxxxx&grant_type=password&username=testuser@xxxxxx.onmicrosoft.com&scope=openid&password=xxxxxxxx
DEBUG [2016-10-07 20:59:46] Response: {"token_type":"Bearer","scope":"user_impersonation","expires_in":"3600","ext_expires_in":"0","expires_on":"1475899186","not_before":"1475895286","resource":"https://management.core.windows.net/","access_token":"xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx","id_token":"yyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyy."}
INFO [2016-10-07 20:59:46] Retrieving tenant info
DEBUG [2016-10-07 20:59:47] Request: https://management.azure.com/tenants?api-version=2015-01-01 (get)
DEBUG [2016-10-07 20:59:47] Response: {"value":[{"id":"/tenants/xxxxxxx","tenantId":"xxxxxxx"}]}
INFO [2016-10-07 20:59:47] Retrieving subscriptions info
DEBUG [2016-10-07 20:59:47] Request: https://management.azure.com/subscriptions?api-version=2015-01-01 (get)
DEBUG [2016-10-07 20:59:47] Response: {"value":[]}
DEBUG [2016-10-07 20:59:47] SubscriptionIDs returned: []
DEBUG [2016-10-07 20:59:47] Credential details: []
INFO [2016-10-07 20:59:47] Creating credentials file at /tmp/tf.creds

INFO [2016-10-07 20:59:47] Done!
```

**Comment by stuartpreston** [comment]
That is strange. From the output it appears it thinks your user (who can authenticate because you get a token in the response) is not associated with a tenant (I guess the same result as you see in the portal)

**Comment by spuder** [comment]
You are right. Initial glance makes one think it is part of a subscription because this user gets the normal azure gui. Only when you try and make a vm does it show you that the user actually isn't part of a subscription 

<img width="647" alt="screenshot 2016-10-07 23 07 16" src="https://cloud.githubusercontent.com/assets/242382/19210533/ebf04ac2-8ce2-11e6-885f-4ca42b1df174.png">


Sadly, I can't add that user to the subscription, despite having the right name. 

<img width="576" alt="screenshot 2016-10-07 23 09 50" src="https://cloud.githubusercontent.com/assets/242382/19210556/4fc45886-8ce3-11e6-9358-7487b2e1fd6a.png">


<img width="513" alt="screenshot 2016-10-07 23 13 53" src="https://cloud.githubusercontent.com/assets/242382/19210584/0163df26-8ce4-11e6-9f86-0eab5ad1a485.png">


Thanks for the tips so far, I've got multiple tickets open with microsoft, and they can't figure it out either. I've also tried your suggestions of using the azure cli, but that path is full of other land mines.
